cmd/kubeadm/app/cmd/images.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"runtime"
+
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+	kuberuntime "k8s.io/apimachinery/pkg/runtime"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	kubeadmapiext "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1alpha1"
+	"k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	"k8s.io/kubernetes/cmd/kubeadm/app/features"
+	"k8s.io/kubernetes/cmd/kubeadm/app/images"
+	"k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/dns"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+)
+
+// NewCmdImages returns the "kubeadm images" command
+func NewCmdImages(out io.Writer) *cobra.Command {
+	cfg := &kubeadmapiext.MasterConfiguration{}
+	legacyscheme.Scheme.Default(cfg)
+	var cfgPath string
+
+	cmd := &cobra.Command{
+		Use:   "images",
+		Short: "Print a list of images kubeadm will use. The configuration file is used in case any images or image repositories are customized.",
+		Run: func(_ *cobra.Command, _ []string) {
+			internalcfg := &kubeadmapi.MasterConfiguration{}
+			legacyscheme.Scheme.Convert(cfg, internalcfg, nil)
+			i, err := NewImages(cfgPath, internalcfg)
+			kubeadmutil.CheckErr(err)
+			kubeadmutil.CheckErr(i.Run(out))
+		},
+	}
+	AddImagesFlags(cmd.PersistentFlags(), &cfgPath)
+
+	return cmd
+}
+
+// NewImages returns a "kubeadm images" command
+func NewImages(cfgPath string, cfg *kubeadmapi.MasterConfiguration) (*Images, error) {
+	if cfgPath != "" {
```

Comment:
Please use `configutil.ConfigFileAndDefaultsToInternalConfig`; this will ensure a consistent initialisation for the master-configuration e.g. by invoking `SetDynamicDefaults` (same as other kubeadm commands)